crates/ordinals/src/brc721/address_mapping.rs

```diff
@@ -0,0 +1,115 @@
+use bitcoin::{Address, Network, WitnessProgram, WitnessVersion};
+use bitcoin_hashes::{hash160::Hash as BTCH160, Hash};
+use sp_core::H160;
+use thiserror::Error;
+
+/// Custom error type for errors related to the address mapping.
+#[derive(Debug, Error, PartialEq)]
+pub enum AddressMappingError {
+	/// Invalid address error
+	#[error("Invalid BTC address: `{0}`. Only P2PKH and P2WPKH supported.")]
+	InvalidAddress(Address),
+}
+
+pub fn btc_address_to_h160(address: Address) -> Result<H160, AddressMappingError> {
```

Comment:
Could the error messaging be more precise? There are many possible errors, but the message is always the same. As a user of this function, it would be nice to know exactly what the error is